Synchronize dom WPT except for dom/events and dom/parts

annevk opened the pull request:
#### 3c7843ecb208897ca867ff064da45d21eb07d41c
<pre>
Synchronize dom WPT except for dom/events and dom/parts
<a href="https://bugs.webkit.org/show_bug.cgi?id=268905">https://bugs.webkit.org/show_bug.cgi?id=268905</a>

Reviewed by Aditya Keerthi.

This is up to and including this commit:
<a href="https://github.com/web-platform-tests/wpt/commit/8359d2b920b1e074a0cf62629fbbb9fd1760de42">https://github.com/web-platform-tests/wpt/commit/8359d2b920b1e074a0cf62629fbbb9fd1760de42</a>

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/dom/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/idlharness-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/idlharness.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-constructor.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-constructor.any.js:
(test):
(test.t.const.source.new.Observable):
(test.t.source.subscribe.complete):
(test.t.source.subscribe.error):
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-constructor.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any.js: Added.
(promise_test.async const):
(promise_test):
(promise_test.async t):
(promise_test.async let):
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.window.js: Added.
(async loadIframeAndReturnContentWindow):
(promise_test.async window):
(promise_test):
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any.js: Added.
(promise_test.async const):
(promise_test):
(promise_test.async t):
(promise_test.async let):
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/dom/slot-recalc-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/large-cdata-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/large-cdata-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/large-cdata.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/resources/large_CDATA.xml: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/resources/large_CDATA_ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/resources/w3c-import.log: Copied from LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/w3c-import.log.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/resources/xml2html.xsl: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/w3c-import.log:
* LayoutTests/tests-options.json:

Canonical link: <a href="https://commits.webkit.org/274273@main">https://commits.webkit.org/274273@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6fc67737909fc8ff9ca41600d0e2e7655004e36d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38292 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17237 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40611 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40850 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/34063 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19998 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14575 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32304 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38865 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14551 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12667 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12648 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34233 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42128 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34856 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34653 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38488 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13229 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10946 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36685 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14784 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/33540 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13639 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5017 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/14247 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders reviewed (commented):
[non-reviewer]

https://github.com/WebKit/WebKit/pull/21912 had some more changes than this, like removing now-gone files from `LayoutTests/tests-options.json`; I forget if that's just a difference with `--clean-dest-dir ` v. not?

Also unclear why you're excluding `dom/events`?

pxlcoder reviewed (approved):
Seems okay to synchronize only what we need.

annevk committed:
Synchronize dom WPT except for dom/events and dom/parts
https://bugs.webkit.org/show_bug.cgi?id=268905

Reviewed by Aditya Keerthi.

This is up to and including this commit:
https://github.com/web-platform-tests/wpt/commit/8359d2b920b1e074a0cf62629fbbb9fd1760de42

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/dom/META.yml:
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/idlharness-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/idlharness.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-constructor.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-constructor.any.js:
(test):
(test.t.const.source.new.Observable):
(test.t.source.subscribe.complete):
(test.t.source.subscribe.error):
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-constructor.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any.js: Added.
(promise_test.async const):
(promise_test):
(promise_test.async t):
(promise_test.async let):
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-takeUntil.window.js: Added.
(async loadIframeAndReturnContentWindow):
(promise_test.async window):
(promise_test):
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any.js: Added.
(promise_test.async const):
(promise_test):
(promise_test.async t):
(promise_test.async let):
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any.worker-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/observable-toArray.any.worker.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/dom/slot-recalc-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/large-cdata-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/large-cdata-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/large-cdata.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/resources/large_CDATA.xml: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/resources/large_CDATA_ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/resources/w3c-import.log: Copied from LayoutTests/imported/w3c/web-platform-tests/dom/observable/tentative/w3c-import.log.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/resources/xml2html.xsl: Added.
* LayoutTests/imported/w3c/web-platform-tests/dom/xslt/w3c-import.log:
* LayoutTests/tests-options.json:

Canonical link: https://commits.webkit.org/274273@main
